core/psh/login.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Phoenix-RTOS
+ *
+ * login - allows/disallows access to Phoenix SHell 
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "psh.h"
+
+
+int psh_login(void){
+	char c, *usrpasswd;
+	unsigned int passsz = 0;
+
+	/* allocate memory for user-given password */
+	if ((usrpasswd = calloc(strlen(psh_common.passwd),sizeof(char)+1)) == NULL )
+	{
+		fprintf(stderr,"psh: fail to allocate memory for login");
+		return -ENOMEM;
+	}
+
+	for (;;)
+	{
```

Comment:
```suggestion
	for (;;) {
```
Please take a look at our coding convention:
https://github.com/phoenix-rtos/phoenix-rtos-doc/blob/master/coding.md